moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -149,19 +149,20 @@ class KinematicsBase
   static const double DEFAULT_SEARCH_DISCRETIZATION; /* = 0.1 */
   static const double DEFAULT_TIMEOUT;               /* = 1.0 */
 
-  /** @brief The signature for a callback that can compute IK */
+  /** @brief Signature for a callback to validate an IK solution */
   typedef boost::function<void(const geometry_msgs::Pose& ik_pose, const std::vector<double>& ik_solution,
                                moveit_msgs::MoveItErrorCodes& error_code)>
       IKCallbackFn;
 
   /**
    * @brief Given a desired pose of the end-effector, compute the joint angles to reach it
+   *
+   * In contrast to the searchPositionIK methods, this one is expected to return the solution
+   * closest to the seed state. Randomly re-seeding is explicitly not allowed.
    * @param ik_pose the desired pose of the link
    * @param ik_seed_state an initial guess solution for the inverse kinematics
    * @param solution the solution vector
    * @param error_code an error code that encodes the reason for failure or success
-   * @param lock_redundant_joints if setRedundantJoints() was previously called, keep the values of the joints marked as
-   * redundant the same as in the seed
```

Comment:
```suggestion
   * @param options An option struct which contains the type of redundancy discretization used. This default
   *                implementation only supports the KinematicSearches::NO_DISCRETIZATION method; requesting any
   *                other will result in failure.
```

Reply:
I don't like to advertize the redundancy parameters anymore. They are not used by MoveIt.
Rather, they are parameters of IKfast only.
However, I added a generic comment as follows:
`@param options container for other IK options. See definition of KinematicsQueryOptions for details.`